### ISSUE TITLE
Featuremodule: neuralnet

### ISSUE THREAD

**Issue by silvanheller** [silvanheller/cineast#5]
The Shotcontainer has a representative frame which we will use. 
Try to abstract the specific tensorflow implementation away from the featuremodule

- [x] Create feature module
- [x] How do we crop & resize the image -> f341b0da452c7c4f4d47e525189817aa42fd6e13
- [x] Support picture input
- [x] Support label input
- [x] What is the behavior for the default constructor of an AbstractFeatureModule since we shouldn't rely on the config?
- [x] Perform Mapping from 300 Entry-level concepts to full 1k labels
- [ ] Code review of feature module
- [x] Refactoring in a way that there are multiple featuremodules per net / training data
- [x] Create neural-net abstraction -> de358e938fb434a31d8f11f1dbe3079bf99294c5
- [ ] Code review of abstraction layers
- [ ] Testing

Wishlist:
- [ ] Detect location [sliding window](https://www.cs.nyu.edu/~fergus/papers/zeilerECCV2014.pdf)
- [ ] Multiple objects [sliding window](https://www.cs.nyu.edu/~fergus/papers/zeilerECCV2014.pdf) or [R-CNN](http://arxiv.org/pdf/1311.2524v5.pdf) 
- [ ] Use an ensemble of neural nets

**Comment by silvanheller** [comment]
See Commit de358e938fb434a31d8f11f1dbe3079bf99294c5 for a first draft of the nn

**Comment by silvanheller** [comment]
We can't use the ResNet [model](https://github.com/ry/tensorflow-resnet) because it requires tensorflow>0.8.0 and maybe also a caffe installation (see [resnet-issue](https://github.com/ry/tensorflow-resnet/issues/2) and [tf-issue](https://github.com/tensorflow/tensorflow/issues/1528)

**Comment by lucaro** [comment]
not true: https://mvnrepository.com/artifact/org.bytedeco.javacpp-presets/tensorflow/0.9.0-1.2

**Comment by silvanheller** [comment]
Thanks, that means the readme of https://github.com/bytedeco/javacpp-presets/tree/master/tensorflow is wrong

**Comment by lucaro** [comment]
Yes, it appears that the readme files in this repository tend to be behind the code in some cases...

**Comment by silvanheller** [comment]
Welcoming Code Review
